src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -98,7 +131,7 @@ bool ClusterResourceScheduler::NodeAlive(scheduling::NodeID node_id) const {
   if (node_id.IsNil()) {
     return false;
   }
-  return is_node_available_fn_(node_id);
+  return is_node_available_fn_ == nullptr || is_node_available_fn_(node_id);
```

Comment:
hmm why would is_node_available_fn_ ever be nullptr?

Reply:
> 

The default constructor of `ClusterResourceScheduler::ClusterResourceScheduler()` does not initialize the `is_node_available_fn_`, I think `is_node_available_fn_` could be initialized to true by default for the time being and then `RAY_CHECK(is_node_available_fn_ != nullptr)` here.